panel/src/helpers/embed.js

```diff
@@ -0,0 +1,131 @@
+export default {
+  youtube(url) {
+
+    if (!url.match("youtu")) {
+      return false;
+    }
+
+    let uri = null;
+
+    try {
+      uri = new URL(url);
+    } catch (e) {
+      return false;
+    }
+
+    const path = uri.pathname.split("/").filter(item => item !== "");
+    const first = path[0];
+    const second = path[1];
+    const host = "https://" + uri.host + "/embed";
+
+    const isYoutubeId = (id) => {
+      return id.match(/^[a-zA-Z0-9_-]+$/) !== null;
+    };
+
+    let query = uri.searchParams;
+    let src = null;
+
+    switch (path.join("/")) {
```

Comment:
When reading the code, I was confused what happens with the query params in the resulting URL, because I didn't yet know they were appended at the end.

Suggestion: A quick comment above this that explains what happens:

```suggestion

    // build the correct base URL for the embed,
    // the query params are appended below
    let src = null;
    switch (path.join("/")) {
```